lib/hsa/mcwamp_hsa.cpp

```diff
@@ -2178,7 +2231,7 @@ class HSAQueue final : public KalmarQueue
 
         std::lock_guard<std::recursive_mutex> lg(qmutex);
 
-        int targetIndex = asyncOp->asyncOpsIndex();
+        int targetIndex = asyncOp->asyncOpsIndex() - asyncOps_offset;
```

Comment:
any statements affected by `asyncOps_offset` deserves to have comment I think

Reply:
Done.